tests/harness/src/case.rs

```diff
@@ -0,0 +1,153 @@
+            .await
+            .with_context(|| format!("Cannot open file {:?}", path.as_ref()))?;
+
+        let mut queries = vec![];
+        let mut query = Query::default();
+
+        let mut lines = BufReader::new(file).lines();
+        while let Some(line) = lines
+            .next_line()
+            .await
+            .with_context(|| format!("Failed to read from file {:?}", path.as_ref()))?
+        {
+            // intercept command start with INTERCEPTOR_PREFIX
+            if line.starts_with(INTERCEPTOR_PREFIX) {
+                query.push_interceptor(line);
+                continue;
+            }
+
+            // ignore comment and empty line
+            if line.starts_with(COMMENT_PREFIX) || line.is_empty() {
+                continue;
+            }
+
+            query.append_query_line(&line);
+
+            // SQL statement ends with ';'
+            if line.ends_with(';') {
+                queries.push(query);
+                query = Query::default();
+            }
+        }
+
+        Ok(Self {
+            name: path.as_ref().to_str().unwrap().to_string(),
+            queries,
+        })
+    }
+
+    pub async fn execute<W>(&self, client: &Client, writer: &mut W) -> Result<()>
+    where
+        W: AsyncWrite + Unpin,
+    {
+        for query in &self.queries {
+            query
+                .execute(client, writer)
+                .await
+                .with_context(|| format!("Error while executing {:?}", query.query_lines))?;
+        }
+
+        Ok(())
+    }
+}
+
+impl Display for TestCase {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&self.name)
+    }
+}
+
+#[derive(Default)]
+struct Query {
+    query_lines: Vec<String>,
+    interceptors: Vec<String>,
+}
+
+impl Query {
+    fn push_interceptor(&mut self, post_process: String) {
+        self.interceptors.push(post_process);
+    }
+
+    fn append_query_line(&mut self, line: &str) {
+        self.query_lines.push(line.to_string());
+    }
+
+    async fn execute<W>(&self, client: &Client, writer: &mut W) -> Result<()>
+    where
+        W: AsyncWrite + Unpin,
+    {
+        let query_ctx = RpcContext {
+            tenant: "public".to_string(),
+            token: "".to_string(),
+        };
+        let query_req = QueryRequest {
+            metrics: vec![],
+            ql: self.build_query_from_lines(),
+        };
+        let result = client.query(&query_ctx, &query_req).await;
+        match result {
+            Ok(rows) => {
+                self.write_result(writer, format!("{}", CsvFormatter { rows }))
+                    .await?
+            }
+            Err(e) => {
+                self.write_result(writer, format!("Failed to execute query, err: {:?}", e))
+                    .await?
+            }
+        }
+
+        Ok(())
+    }
+
+    // TODO: implement query interceptor.
+    fn intercept(&self, _result: QueriedRows) -> QueriedRows {
```

Comment:
This intercept is used for post process? If yes, in what situation would users need this(instead of using SQL function to archive same result)?

Reply:
E.g:
- require the runner to sort the result
- replace some non-constant, like timestamp
- display something out to the output

Some SQL's results won't always be the same and it isn't required to be the same. Like query without sort, or gets the current timestamp. This interceptor performs some pre- or post-process that makes us can still compare the result

Reply:
> require the runner to sort the result
> display something out to the output
> gets the current timestamp

Those can be done in SQL itself.

> Like query without sort

In this situation, what should I do in post-process in order to make it comparable?

FYI, I just think an easy-to-use post-process framework is not easy to design, if we have no clear requirements, we can think about it later.

Reply:
How about this syntax? 
```
-- postprocess
```
IMO It's more clear than before.

Reply:
Preprocess like changing settings is also allowed.

Reply:
Then like this?
```
-- interceptor
```

Reply:
this might be too general. The preserved "keyword" should be more special?

Reply:
I'm fine this `keyword` if we haven't get a better one...